pallets/pools/src/lib.rs

```diff
@@ -619,9 +630,7 @@ pub mod pallet {
 		pub fn update(
 			origin: OriginFor<T>,
 			pool_id: T::PoolId,
-			min_epoch_time: u64,
-			challenge_time: u64,
-			max_nav_age: u64,
+			changes: PoolUpdate<T::InterestRate>,
```

Comment:
Instead of having single update function, I would split it into multiple small update functions. Cheaper and easier to extend and we don`t have the changes struct.

Reply:
I think this would complicate the logic quite a lot. Would it be a `Vec` of multiple potential changes, with separate timestamps? I think the use case is that issuers won't do updates like this often (maybe a few times per year), and will likely combine multiple changes.

Reply:
I am not sure about this though. Anytime, we add some updatable settings to the pools, the API of this changes. Splitting it into multiple smaller update-extrinsics is relatively stable. 

But if we think, the updatable settings will be stable, I would be fine with this.

Reply:
We could think abou the `Updates` storage holding a `Vec<PoolUpdates>` where `PoolUpdates` is something like:

```rust
enum PoolUpdates {
    MinEpochTime(Moment),
    ....
}
```

Or we have separate storage elements for each updatable setting in the pool.

Both solutions allow us to extend these settings without a storage migration (assuming if we only append the enum and do not remove stuff)